tests/test_collapse.py

```diff
@@ -76,13 +74,51 @@ def test_reconcile_prefix(self):
         """Test curie reconciliation is performing as expected."""
         msdf = parse_sssom_table(data_dir / "basic3.tsv")
 
-        with open(prefix_recon_yaml) as pref_rec:
-            prefix_reconciliation = yaml.safe_load(pref_rec)
+        self.assertEqual(
+            {
+                "a": "http://example.org/a/",
+                "b": "http://example.org/b/",
+                "c": "http://example.org/c/",
+                "d": "http://example.org/d/",
+                "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
+                "owl": "http://www.w3.org/2002/07/owl#",
+                "orcid": "https://orcid.org/",
+                "semapv": "https://w3id.org/semapv/vocab/",
+                "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+                "skos": "http://www.w3.org/2004/02/skos/core#",
+                "sssom": "https://w3id.org/sssom/",
+            },
+            msdf.prefix_map,
+        )
+        prefix_reconciliation = {
+            "prefix_synonyms": {
+                "a": "c",
+                "b": "bravo",
+                "r": "rdfs",
+                "o": "owl",
+            },
+            "prefix_expansion_reconciliation": {
+                "c": "http://test.owl/c/",
+                "bravo": "http://test.owl/bravo",
+                "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
+                "owl": "http://www.w3.org/2002/07/owl#",
+            },
+        }
 
         recon_msdf = reconcile_prefix_and_data(msdf, prefix_reconciliation)
-
-        prefix_expansion = prefix_reconciliation["prefix_expansion_reconciliation"]
-
-        for pfx, exp in prefix_expansion.items():
-            if pfx in recon_msdf.prefix_map.keys():
-                self.assertEqual(recon_msdf.prefix_map[pfx], exp)
+        self.assertEqual(
+            {
+                "a": "http://example.org/a/",
```

Comment:
Why is "a" still present? Shouldn't it have been remapped to c before it go rewired?

Reply:
The mapping "a":"c" got ignored since `c` is already present in the prefix map and this would cause an inconsistency

Reply:
So the question is: do we want to add an "overwrite" setting? I don't think that having this be default functionality makes sense. if you don't want original prefix `c` in the map, then maybe there should be a 3rd part to this workflow for explicit deletions

Reply:
Hmmm interesting. I think we would like a --force option, which results in the removal of the already existing entry, and a logging.warning(). What do you think?

Reply:
Yes, I think having a flag would be the way to go. I will mock that in the `curies` package.

Reply:
I have a solution! Pushing now

Reply:
Seems to work nicely! My new solution doesn't allow deletions, but requires that if you remap a->c and there is already a c, you need to give c-> something else. Seems reasonable to me